Mathlib/NumberTheory/Padics/WithVal.lean

```diff
@@ -0,0 +1,181 @@
+/-
+Copyright (c) 2025 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+import Mathlib.Analysis.RCLike.Basic
+import Mathlib.GroupTheory.ArchimedeanDensely
+import Mathlib.NumberTheory.Padics.PadicNumbers
+import Mathlib.Topology.Algebra.UniformRing
+import Mathlib.Topology.Algebra.Valued.ValuedField
+import Mathlib.Topology.Algebra.Valued.WithVal
+import Mathlib.Topology.GDelta.MetrizableSpace
+
+/-!
+# Equivalence between `ℚ_[p]` and `WithVal v`
```

Comment:
Can you expand this doc a bit more, saying what `v` is and detailling what equivalences (i.e. as what) are proven in the file?